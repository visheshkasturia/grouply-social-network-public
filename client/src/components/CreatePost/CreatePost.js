/*eslint-disable */
/* istanbul ignore file */
import { useState } from "react";
import PostPosts from "../../apiCalls/PostPosts";
import { uploadMedia } from "../../apiCalls/uploadMedia";

export default function CreatePost( {user, updatePostInGroup} ) {
  const [mediaURL, setMediaURL] = useState('');
  const [inputs, setInputs] = useState({
    postTitle: '',
    postContent: '',
    hashTags: '',
    partOf: '',
  });
  const handleChange = (event) => {
    const name = event.target.name;
    const value = event.target.value;
    setInputs(values => ({...values, [name]: value}))
  }


  const handleSubmit = async (event) => {
    event.preventDefault();
    inputs.createdBy = user._id;
    if (mediaURL !== '') {
      inputs.postContent = mediaURL;
    }
    await PostPosts(inputs);
    setInputs({
      postTitle: '',
      postContent: '',
      hashTags: '',
      partOf: '',
    });
    setMediaURL('')
    updatePostInGroup();
  }

  async function uploadImage(event) {
    const file = event.target.files[0];
    if (file) {
      const imageURL = await uploadMedia({
        type: 'image',
        file
      });
      setMediaURL(imageURL);
    }
  }

  async function uploadVideo(event) {
    const file = event.target.files[0];
    if (file) {
      const videoURL = await uploadMedia({
        type: 'video',
        file
      });
      setMediaURL(videoURL);
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <label>Enter Post Title:
      <input 
        type="text" 
        name="postTitle" 
        value={inputs.postTitle} 
        onChange={handleChange}
      />
      </label>
      <label>Enter Post Content:
      <textarea 
        rows="5"
        cols="40"
        name="postContent" 
        value={inputs.postContent } 
        onChange={handleChange}
      />
      </label>
     <label>Enter Post hashtags:
        <input 
          type="text" 
          name="hashTags" 
          value={inputs.hashTags} 
          onChange={handleChange}
        />
     </label>
     <label>Enter Group Name you want to Post in:
        <input 
          type="text" 
          name="partOf" 
          value={inputs.partOf} 
          onChange={handleChange}
        />
     </label>

     <label> Upload Image </label>
     <input 
      type='file'
      onChange={uploadImage}
      id='imageUpload'
      />
     
    <label> Upload Audio/Video </label>
    <input 
      type='file'
      onChange={uploadVideo}
      id='VideoUpload'
    />

      <br/> 
      <button type="submit"> Submit </button>
    </form>
  )
}