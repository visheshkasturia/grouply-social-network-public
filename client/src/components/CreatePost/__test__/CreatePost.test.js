import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CreatePost from '../CreatePost'

it('CreatePost renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CreatePost />, div);
});

it('CreatePost matches snapshot', () => {
  const tree = renderer.create(<CreatePost />).toJSON();
  expect(tree).toMatchSnapshot();
});