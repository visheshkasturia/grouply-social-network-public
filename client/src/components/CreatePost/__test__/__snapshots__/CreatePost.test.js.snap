// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePost matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Enter Post Title:
    <input
      name="postTitle"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label>
    Enter Post Content:
    <textarea
      cols="40"
      name="postContent"
      onChange={[Function]}
      rows="5"
      value=""
    />
  </label>
  <label>
    Enter Post hashtags:
    <input
      name="hashTags"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label>
    Enter Group Name you want to Post in:
    <input
      name="partOf"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label>
     Upload Image 
  </label>
  <input
    id="imageUpload"
    onChange={[Function]}
    type="file"
  />
  <label>
     Upload Audio/Video 
  </label>
  <input
    id="VideoUpload"
    onChange={[Function]}
    type="file"
  />
  <br />
  <button
    type="submit"
  >
     Submit 
  </button>
</form>
`;
